platform/mk2/libs/STM32_USB-Host-Device_Lib_V2.1.0/Libraries/STM32_USB_Device_Library/Class/cdc/src/usbd_cdc_core.c

```diff
@@ -222,8 +222,8 @@ __ALIGN_BEGIN uint8_t usbd_cdc_CfgDesc[USB_CDC_CONFIG_DESC_SIZ]  __ALIGN_END = {
     0x02,   /* bNumInterfaces: 2 interface */
     0x01,   /* bConfigurationValue: Configuration value */
     0x00,   /* iConfiguration: Index of string descriptor describing the configuration */
-    0x80,   /* bmAttributes: bus powered */
-    0x32,   /* MaxPower 0 mA */
+    0xC0,   /* bmAttributes: self powered */

```

Comment:
Self powered?  I thought we agreed on Bus powered since we still need to support the case where users use USB power to program the device.  The logic being that if the device doesn't actually use all 500mA of power, its no big deal (IIRC the spec even says this is ok).